### ISSUE TITLE
Details sidebar accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7308]
### Description
The details sidebar can be found on the right side of several edit views: assets, entries, categories, users. There may be more instances to identify.
![Screenshot of details sidebar.](https://user-images.githubusercontent.com/17884198/102730417-93d5af00-42e9-11eb-943c-25bd25292def.png)

Accessibility notes for this area:
- All: The editable fields don't have a border around them, so it's not clear that they are editable form fields. This is especially the case when there's only one editable field and no close/x icons nearby, such as a "filename" for assets:
![Screenshot of Filename field.](https://user-images.githubusercontent.com/17884198/102730702-6dfcda00-42ea-11eb-904b-adfaecc0a255.png)
- All: It would be helpful to label the top and bottom boxes. Not sure of good headings yet, but something that indicates the top section is editable and the bottom section is informative.
- Entries: The "Notes about your changes" text box has insufficient contrast and is using a placeholder as a label.
- Users: The delete functionality is inconsistent with the other editable views. It is located under a gear icon menu rather than within the save menu.
![Screenshot of user delete option.](https://user-images.githubusercontent.com/17884198/102730800-c16f2800-42ea-11eb-96f6-8c761df46287.png)
![Screenshot of category delete option.](https://user-images.githubusercontent.com/17884198/102730809-c9c76300-42ea-11eb-97af-a85a80cd0054.png)

### Steps to reproduce
1. Edit or create a new asset, entry, category, or user.

### Additional info
- Craft version: 3.5.17.1

**Comment by ajanec01** [comment]
> h5 elements are causing heading level errors. I think these don't necessarily need to be headings. Maybe change to definition lists.
+1 for `<dl>`- that was also my first impression

**Comment by ajanec01** [comment]
Some other issues that I've found when checking assets entry

- the choose button and the switch have poor focus indication (I know you're going to have a new focus styling, let me know if you need the instances to be logged or should I not bother raising them)
- just a side note about the author element not being operable, the author does not have an acc role and it should not be a container for the remove link (needs href)
![author-button](https://user-images.githubusercontent.com/47136246/103714024-9147aa80-4fb5-11eb-9421-e5b7c5851a7b.png)
- the author pop-up would benefit from `role="dialog"` and an accessible name; the focus indicator is poor and it's not trapped inside the popup; the content would benefit from the `<dl>` markup
![author-popup](https://user-images.githubusercontent.com/47136246/103714108-c0f6b280-4fb5-11eb-876e-1e25600887e1.png)
- The post date, post time, expiry date, and expiry time are not labelled; ideally, the label/ description would include the entry format (e.g. dd/mm/yyyy); there are no error messages for giving an incorrect format, I've tried 23/121/2020 and the entry got saved with today's date
![date-and-time-format](https://user-images.githubusercontent.com/47136246/103714678-e3d59680-4fb6-11eb-9597-ec7d9d391efa.png)
- date and time pickers are not keyboard accessible; worth having a look at the [aria date picker](https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/datepicker-dialog.html) and [aria listbox](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-collapsible.html) patterns.
- Finally, I wasn't too sure what format the post date and expiry date fields should have as they have more elements than any other; I've checked the spec for `<dl>` and it's appropriate to have [one `dt` and a few `dd`](https://html.spec.whatwg.org/#the-dl-element:the-dt-element-8); I didn't know that earlier so I thought I would raise it just in case
